### PR TITLE
Some assorted spotfinding cleanup and light refactor

### PR DESCRIPTION
**Pull request by ndevenish**
I've been working on some refactoring in the spotfinding code. It's somewhat stuck in tar, but some of the base-level cleaning is probably useful to put in now. This is here in this commit.

- Using the `SpotFinder` class has been changed from a `__call__` call to `SpotFinder.find_spots` - to make it more obvious what is going on. Because this is somewhat module-external API, have deprecated instead of outright removal of the old interface
- `PixelListToShoeboxes`, `ShoeboxesToReflectionTable` and `PixelListToReflectionTable` Functor-classes have been converted to functions  - `pixel_list_to_shoeboxes`, `shoeboxes_to_reflection_table` and `pixel_list_to_reflection_table`
- Added a `__repr__` for `PixelList` of form `"<PixelList frame=10 size=100,100 len=4989>"`
- Done some local cleanup in `finder.py` and `factory.py` - python3 upgrading, floating, moving to logger-based formatting for log messages. Flake8 warnings fixed.

- Fix bug where logger messages from multiprocessing workers were not being formatted correctly

### CODE REVIEW

**Comment by graeme-winter** [comment]
@ndevenish `stdout` from dials.find_spots now contains 

```
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
Found %d strong pixels on image %d
```

despite apparently working 🙃

**Comment by graeme-winter** [comment]
Timing: 900 image `gz` compressed small molecule data set; on branch:

```
Histogram of per-image spot count for imageset 0:
23070 spots found on 900 images (max 476 / bin)
         *  *   *         *                                 
*** * ** *************** ** *   * *                         
************************************** *    *              *
******************************************* **** ***** *****
************************************************************
************************************************************
************************************************************
************************************************************
************************************************************
************************************************************
1                         image                          900

--------------------------------------------------------------------------------
Saved 23070 reflections to strong.refl

real	0m41.568s
user	2m15.459s
sys	0m12.637s
```

on master:

```
Histogram of per-image spot count for imageset 0:
23070 spots found on 900 images (max 476 / bin)
         *  *   *         *                                 
*** * ** *************** ** *   * *                         
************************************** *    *              *
******************************************* **** ***** *****
************************************************************
************************************************************
************************************************************
************************************************************
************************************************************
************************************************************
1                         image                          900

--------------------------------------------------------------------------------
Saved 23070 reflections to strong.refl

real	0m41.290s
user	2m13.827s
sys	0m12.582s
```

(having already run spot finding once to cache the images)

FWIW yes the numbers are included in the `stdout` on master

```
...
Found 1058 strong pixels on image 680
Found 1088 strong pixels on image 681
Found 1188 strong pixels on image 682
Found 1294 strong pixels on image 683
Found 1198 strong pixels on image 684
Found 1049 strong pixels on image 685
...
```

File: algorithms/spot_finding/factory.py [hunk 1]
```diff
@@ -395,13 +393,11 @@ def __call__(self, flags, **kwargs):
         num_before = flags.count(True)
         flags = self.run(flags, **kwargs)
         num_after = flags.count(True)
-        logger.info(
-            "Filtered {0} of {1} spots by spot density".format(num_after, num_before)
-        )
+        logger.info("Filtered %d of %d spots by spot density", num_after, num_before)
```
Comment (graeme-winter) [inline algorithms/spot_finding/factory.py:396]: This is not a good change as the numbers don't appear in the output 🙂

**Review by graeme-winter — APPROVED** [review]
Not looked at the diffs but I would assume given the author that these are better than I would do. I only observe a change which appears to have the effect of removing useful information from the `stdout` so approving modulo that being fixed.

File: algorithms/spot_finding/factory.py (hunk 1 above)
Comment (graeme-winter) [reply algorithms/spot_finding/factory.py:396]: This comment is wrong 🤦‍♂️

That line _does_ contain numbers in stdout

**Comment by ndevenish** [comment]
I'm slightly baffled by this because it appears to be valid
```python
import logging
logger = logging.getLogger()
logging.basicConfig(level=logging.DEBUG)
logging.info("info %d %d", 1, 2)
```
Gives `INFO:root:info 1 2`

**Comment by graeme-winter** [comment]
> I'm slightly baffled by this because it appears to be valid
> 
> ```python
> import logging
> logger = logging.getLogger()
> logging.basicConfig(level=logging.DEBUG)
> logging.info("info %d %d", 1, 2)
> ```
> 
> Gives `INFO:root:info 1 2`

only an issue if `nproc>1`

**Comment by graeme-winter** [comment]
@ndevenish verify that this is now working correctly

**Comment by graeme-winter** [comment]
(cosmetic) banner looking weird now though

```
Setting spotfinder.filter.min_spot_size=3
Configuring spot finder from input parameters
--------------------------------------------------------------------------------Finding strong spots in imageset 0--------------------------------------------------------------------------------
```

**Comment by graeme-winter** [comment]
Should look like 

```

Setting spotfinder.filter.min_spot_size=3
Configuring spot finder from input parameters
--------------------------------------------------------------------------------
Finding strong spots in imageset 0
--------------------------------------------------------------------------------
```

**Comment by graeme-winter** [comment]
👍  

output all good now thank you - good to go

**Pull request merged**